service/worker/scheduler/workflow.go

```diff
@@ -1027,14 +1062,13 @@ func (s *scheduler) updateMemoAndSearchAttributes() {
 		}
 	}
 
-	currentPausedPayload := workflowInfo.SearchAttributes.GetIndexedFields()[searchattribute.TemporalSchedulePaused]
-	var currentPaused bool
-	if currentPausedPayload == nil ||
-		payload.Decode(currentPausedPayload, &currentPaused) != nil ||
-		currentPaused != s.Schedule.State.Paused {
-		err := workflow.UpsertSearchAttributes(s.ctx, map[string]interface{}{
-			searchattribute.TemporalSchedulePaused: s.Schedule.State.Paused,
-		})
+	schedulePausedKey := temporal.NewSearchAttributeKeyBool(searchattribute.TemporalSchedulePaused)
+	currentPaused, exists := workflow.GetTypedSearchAttributes(s.ctx).GetBool(schedulePausedKey)
+	if !exists || currentPaused != s.Schedule.State.Paused {
+		err := workflow.UpsertTypedSearchAttributes(
+			s.ctx,
+			schedulePausedKey.ValueSet(s.Schedule.State.Paused),
+		)
```

Comment:
are you 100% sure that this is compatible with the previous code? note that if the previous code decided to do an upsert and this does not, or vice versa, then it's a determinism violation. if there's any doubt I'd prefer to just leave the old code

Reply:
Gonna revert.